hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/BaseRollbackHelper.java

```diff
@@ -116,8 +118,13 @@ List<Pair<String, HoodieRollbackStat>> maybeDeleteAndCollectStats(HoodieEngineCo
                                                                     HoodieInstant instantToRollback,
                                                                     List<SerializableHoodieRollbackRequest> rollbackRequests,
                                                                     boolean doDelete, int numPartitions) {
+    // The rollback requests for append only exist in table version 6 and below which require groupBy
+    List<SerializableHoodieRollbackRequest> processedRollbackRequests =
+        metaClient.getTableConfig().getTableVersion().greaterThanOrEquals(HoodieTableVersion.EIGHT)
+            ? rollbackRequests
```

Comment:
Does the issue exist for 0.x release, is it a logic miss for compatibility migration?

Reply:
This is an issue in 0.x release or table version 6.  If there are multiple log files in the same file group to be rolled back in the rollback request, the existing rollback logic concurrently executes them (appending rollback log blocks in table version 6), which can cause issue of marker creation failure, i.e., both of the Spark executor tasks try to create the log file with the same name.